### ISSUE TITLE
Can't run games with 2.9.0

### ISSUE THREAD

**Issue by ripsivis** [Heroic-Games-Launcher/HeroicGamesLauncher#2902]
### Describe the bug

After applying the 2.9.0 update, I couldn't be able to run the games via Heroic client. I did run the games through their executables successfully. I tried resetting Heroic, doing a clean install but no luck. I downgraded to 2.8.0 and client now can run games as it should.

### Add logs

```text
(19:36:02) INFO:    [Legendary]:        Running command: XDG_CONFIG_HOME=C:\Users\User1\AppData\Roaming\heroic\legendaryConfig C:\Users\User1\AppData\Local\Programs\heroic\resources\app.asar.unpacked\build\bin\win32\legendary --version
(19:36:02) INFO:    [Legendary]:        Legendary location: C:\Users\User1\AppData\Local\Programs\heroic\resources\app.asar.unpacked\build\bin\win32\legendary
(19:36:02) INFO:    [Gog]:              GOGDL location: C:\Users\User1\AppData\Local\Programs\heroic\resources\app.asar.unpacked\build\bin\win32\gogdl
(19:36:02) INFO:    [Connection]:       Connectivity: check-online
(19:36:02) INFO:    [Connection]:       Pinging external endpoints
(19:36:02) INFO:    [Backend]:          DRM module staus {
  "neifaoindggfcjicffkgpmnlppeffabd": {
    "name": "Google Widevine Windows CDM",
    "status": "new",
    "version": "1.0.2512.0"
  },
  "oimompecagnajdejgnnjijobebaeigek": {
    "name": "Widevine Content Decryption Module",
    "status": "new",
    "version": "4.10.2557.0"
  }
}
(19:36:03) WARNING: [Backend]:          Protocol already registered.
(19:36:03) INFO:    [Connection]:       Connectivity: online
(19:36:03) ERROR:   [Gog]:              Unable to syncQueued playtime, userData not present
(19:36:03) INFO:    [Backend]:          VCRuntime is installed
(19:36:03) INFO:    [Gog]:              Running command: C:\Users\User1\AppData\Local\Programs\heroic\resources\app.asar.unpacked\build\bin\win32\gogdl --auth-config-path C:\Users\User1\AppData\Roaming\heroic\gog_store\auth.json --version
(19:36:03) INFO:    [Backend]:          Loading Screen Ready
(19:36:03) INFO:    [Frontend]:         Refreshing undefined Library
(19:36:03) WARNING: [Backend]:          refresh not implemented on Sideload Library Manager
(19:36:03) INFO:    [Legendary]:        Refreshing library...
(19:36:03) INFO:    [Legendary]:        Refreshing Epic Games...
(19:36:03) INFO:    [Legendary]:        Game list updated, got 37 games & DLCs
(19:36:03) INFO:    [Backend]:          Frontend Ready
(19:36:03) INFO:    [ProtocolHandler]:  received 'heroic://launch/legendary/9d2d0eb64d5c44529cece33fe2a46482'
(19:36:03) WARNING: [Backend]:          listUpdateableGames not implemented on Sideload Library Manager
(19:36:03) INFO:    [Gog]:              Found 0 game(s) to update
(19:36:03) INFO:    [Backend]:          Launching Grand Theft Auto V (9d2d0eb64d5c44529cece33fe2a46482)
(19:36:03) INFO:    [Backend]:          Preventing display from sleep
(19:36:03) INFO:    [Legendary]:        Running command: XDG_CONFIG_HOME=C:\Users\User1\AppData\Roaming\heroic\legendaryConfig C:\Users\User1\AppData\Local\Programs\heroic\resources\app.asar.unpacked\build\bin\win32\legendary --version
(19:36:03) INFO:    [Backend]:          Preventing machine to sleep
(19:36:03) INFO:    [Backend]:          Stopping Power Saver Blocker
(19:36:03) INFO:    [Backend]:          Checking for new Heroic Updates
(19:36:03) INFO:    [Backend]:          Checking for current version changelog
(19:36:03) INFO:    [Legendary]:        Running command: XDG_CONFIG_HOME=C:\Users\User1\AppData\Roaming\heroic\legendaryConfig C:\Users\User1\AppData\Local\Programs\heroic\resources\app.asar.unpacked\build\bin\win32\legendary list --third-party
(19:36:03) INFO:    [Legendary]:        Checking for game updates: XDG_CONFIG_HOME=C:\Users\User1\AppData\Roaming\heroic\legendaryConfig C:\Users\User1\AppData\Local\Programs\heroic\resources\app.asar.unpacked\build\bin\win32\legendary list --third-party
(19:36:03) INFO:    [Legendary]:        Launching Grand Theft Auto V: C:\Users\User1\AppData\Local\Programs\heroic\resources\app.asar.unpacked\build\bin\win32\legendary launch 9d2d0eb64d5c44529cece33fe2a46482 --language en --wrapper
(19:36:04) INFO:    [Nile]:             Running command: XDG_CONFIG_HOME=C:\Users\User1\AppData\Roaming\heroic\nile_config C:\Users\User1\AppData\Local\Programs\heroic\resources\app.asar.unpacked\build\bin\win32\nile --version
(19:36:04) INFO:    [Gog]:              Running command: C:\Users\User1\AppData\Local\Programs\heroic\resources\app.asar.unpacked\build\bin\win32\gogdl --auth-config-path C:\Users\User1\AppData\Roaming\heroic\gog_store\auth.json --version
(19:36:04) INFO:    [Backend]:          Stopping Display Power Saver Blocker
(19:36:05) INFO:    [Nile]:             Running command: XDG_CONFIG_HOME=C:\Users\User1\AppData\Roaming\heroic\nile_config C:\Users\User1\AppData\Local\Programs\heroic\resources\app.asar.unpacked\build\bin\win32\nile --version
(19:36:05) DEBUG:   [Backend]:          Unable to load Steam Libraries, libraryfolders.vdf not found
(19:36:06) INFO:    [Legendary]:        Found 0 games to update
(19:36:08) WARNING: [Backend]:          Could not determine System Info
(19:36:08) INFO:    [Backend]:          Starting the Download Queue
(19:36:08) WARNING: [Backend]:          Could not determine System Info
(19:36:16) INFO:    [Backend]:          Launching Control (Calluna)
(19:36:16) INFO:    [Backend]:          Preventing display from sleep
(19:36:16) INFO:    [Legendary]:        Running command: C:\Users\User1\AppData\Local\Programs\heroic\resources\app.asar.unpacked\build\bin\win32\legendary --version
(19:36:16) INFO:    [Backend]:          Stopping Power Saver Blocker
(19:36:17) INFO:    [Legendary]:        Launching Control: C:\Users\User1\AppData\Local\Programs\heroic\resources\app.asar.unpacked\build\bin\win32\legendary launch Calluna --language en --wrapper
(19:36:17) INFO:    [Gog]:              Running command: C:\Users\User1\AppData\Local\Programs\heroic\resources\app.asar.unpacked\build\bin\win32\gogdl --auth-config-path C:\Users\User1\AppData\Roaming\heroic\gog_store\auth.json --version
(19:36:17) INFO:    [Backend]:          Stopping Display Power Saver Blocker
(19:36:17) INFO:    [Frontend]:         Refreshing legendary Library
(19:36:17) INFO:    [Legendary]:        Refreshing library...
(19:36:17) INFO:    [Legendary]:        Refreshing Epic Games...
(19:36:17) INFO:    [Legendary]:        Game list updated, got 37 games & DLCs
(19:36:17) INFO:    [Nile]:             Running command: XDG_CONFIG_HOME=C:\Users\User1\AppData\Roaming\heroic\nile_config C:\Users\User1\AppData\Local\Programs\heroic\resources\app.asar.unpacked\build\bin\win32\nile --version
(19:36:17) INFO:    [Legendary]:        Running command: C:\Users\User1\AppData\Local\Programs\heroic\resources\app.asar.unpacked\build\bin\win32\legendary list --third-party
(19:36:22) WARNING: [Backend]:          Could not determine System Info
```

### Steps to reproduce

1. Start the client.
2. Try to run GTAV and Control

### Expected behavior

- Execute the games.

### Heroic Version

Latest Stable

### System Information

- OS : Windows 11 22H2

**Comment by Virusnest** [comment]
Same thing is happening to me on MacOS and games also wont install!

**Comment by Virusnest** [comment]
This also happens to me on MacOS I get the same logs as those before it fails to launch.

**Comment by kturques** [comment]
Do you guys see a log in this format: GAMENAME-lastPlay.log?
On Linux it is inside the folder /home/USERNAME/.config/**heroic/GamesConfig/**
For me it shows the real error, like this:
**legendary launch: error: argument --wrapper: expected one argument**

**Comment by dnahogyil** [comment]
> Do you guys see a log in this format: GAMENAME-lastPlay.log? On Linux it is inside the folder /home/USERNAME/.config/**heroic/GamesConfig/** For me it shows the real error, like this: **legendary launch: error: argument --wrapper: expected one argument**

Game Log:
usage: legendary launch <App Name> [options]
legendary launch: error: argument --wrapper: expected one argument

Confirmed same error

**Comment by TheNando** [comment]
I'm having this issue in Windows with version 2.9.0. I can see in the logs that the command being run in `<my user dir>\gogdl.exe launch "D:\Games\Gravity Circuit" 1201648069 --wrapper  --platform windows` and just like @dnahogyil points out, the error is `--wrapper: expected one argument`. Its looks like this wrapper arg is being included with an empty value. If I run the same command in my terminal without the `--wrapper` the game loads just fine.

**Comment by TheNando** [comment]
It looks like the `--wrapper` issue has been addressed in the most recent commit (as of the time I'm writing this)
https://github.com/Heroic-Games-Launcher/HeroicGamesLauncher/commit/5212c78d32a53709d33e2e5e22a2b28db7b0caf3

**Comment by siyman** [comment]
> It looks like the `--wrapper` issue has been addressed in the most recent commit (as of the time I'm writing this) [5212c78](https://github.com/Heroic-Games-Launcher/HeroicGamesLauncher/commit/5212c78d32a53709d33e2e5e22a2b28db7b0caf3)

Any possibility to test this out without building it locally as I am having the same issue?

**Comment by imLinguin** [comment]
You can get build from actions that run on every commit. Please note those releases are not signed.

![obraz](https://github.com/Heroic-Games-Launcher/HeroicGamesLauncher/assets/62100117/23ba4bdb-f1be-4f73-87f8-f699b4b6f38b)

**Comment by Francehelder** [comment]
But, how can download this?

**Comment by ripsivis** [comment]
> Do you guys see a log in this format: GAMENAME-lastPlay.log? On Linux it is inside the folder /home/USERNAME/.config/**heroic/GamesConfig/**
> 
> If the error in this log is something like this: "legendary launch: error: argument --wrapper: expected one argument", it's the same bug reported here: #2895

No 

![image](https://github.com/Heroic-Games-Launcher/HeroicGamesLauncher/assets/57156687/f8bac11f-a3e6-4572-bee0-a15672c4d4fd)

**Comment by kturques** [comment]
@ripsivis 

Maybe in this folder:
C:\Users\YOUR_USERNAME\Appdata\Roaming\heroic\GamesConfig\

**Comment by kturques** [comment]
> But, how can download this?

Click in **Details**, in the next page click in **Summary**
Bottom of the page, in **artifacts**, choose according to your platform

**Comment by oldlamps** [comment]
Recompiled from GIT after https://github.com/Heroic-Games-Launcher/HeroicGamesLauncher/commit/5212c78d32a53709d33e2e5e22a2b28db7b0caf3 and everything is installing and launching as it should, including Amazon games

**Comment by profiluefter** [comment]
> Recompiled from GIT after [5212c78](https://github.com/Heroic-Games-Launcher/HeroicGamesLauncher/commit/5212c78d32a53709d33e2e5e22a2b28db7b0caf3) and everything is installing and launching as it should, including Amazon games

I can also confirm that the new build works for me on Windows 10!

**Comment by siyman** [comment]
Yip, thanks to the latest Wine-GE and the fix Amazon Games are up and running for me, too.

**Comment by arielj** [comment]
Closing since it's a known fix of 2.9.1
